lib/mongo/uri.rb

```diff
@@ -335,11 +335,31 @@ def database
     #
     # @return [ String ] The uri string.
     def to_s
-      @string
+      reconstruct_uri
     end
 
     private
 
+    # Reconstruct the URI from its parts.
```

Comment:
So this should say that invalid option values are dropped and option names are converted to lower|canonical case.

Reply:
done